dbt/adapters/bigquery/dataproc/batch.py

```diff
@@ -0,0 +1,63 @@
+from typing import Union, Dict
+
+import time
+from datetime import datetime
+from google.cloud.dataproc_v1 import (
+    CreateBatchRequest,
+    BatchControllerClient,
+    Batch,
+    GetBatchRequest,
+)
+from google.protobuf.json_format import ParseDict
+
+from dbt.adapters.bigquery.connections import DataprocBatchConfig
+
+_BATCH_RUNNING_STATES = [Batch.State.PENDING, Batch.State.RUNNING]
+DEFAULT_JAR_FILE_URI = "gs://spark-lib/bigquery/spark-bigquery-with-dependencies_2.12-0.21.1.jar"
+
+
+def create_batch_request(batch: Batch, batch_id, project, region) -> CreateBatchRequest:
+    return CreateBatchRequest(
+        parent=f"projects/{project}/locations/{region}",  # type: ignore
+        batch_id=batch_id,
+        batch=batch,  # type: ignore
+    )
+
+
+def poll_batch_job(parent, batch_id, job_client: BatchControllerClient, timeout: int) -> Batch:
+    batch_name = "".join([parent, "/batches/", batch_id])
+    state = Batch.State.PENDING
+    response = None
+    run_time = 0
+    while state in _BATCH_RUNNING_STATES and run_time < timeout:
+        time.sleep(1)
+        response = job_client.get_batch(  # type: ignore
+            request=GetBatchRequest(name=batch_name),  # type: ignore
+        )
+        run_time = datetime.now().timestamp() - response.create_time.timestamp()  # type: ignore
+        state = response.state
```

Comment:
Do we need to check if `response` is `None` here? We're throwing an error for that condition after we exit the while loop, so it seems like a possibility. This would throw an attribute error in that case, prior to getting to the error check.

Same issue with `response.create_time.timestamp()` above.

Reply:
No we don't need to check here since the call will return a response or raise an exception

Reply:
tbh the `if not response` check isn't _really_ needed but it makes the type checker happy